pkg/agent/nodeportlocal/k8s/npl_controller.go

```diff
@@ -203,25 +203,31 @@ func (c *NPLController) enqueueSvcUpdate(oldObj, newObj interface{}) {
 	}
 
 	podKeys := sets.String{}
+	var oldPodSet, newPodSet sets.String
 	if oldSvcAnnotation != newSvcAnnotation {
 		// Process Pods corresponding to Service with valid NPL annotation.
 		if oldSvcAnnotation == "true" {
-			podKeys = sets.NewString(c.getPodsFromService(oldSvc)...)
+			oldPodSet = sets.NewString(c.getPodsFromService(oldSvc)...)
 		} else if newSvcAnnotation == "true" {
-			podKeys = sets.NewString(c.getPodsFromService(newSvc)...)
+			newPodSet = sets.NewString(c.getPodsFromService(newSvc)...)
 		}
 	}
 
-	var oldPodSet, newPodSet sets.String
 	if oldSvc.Spec.Type != newSvc.Spec.Type {
-		newPodSet = sets.NewString(c.getPodsFromService(newSvc)...)
+		if newPodSet != nil {
```

Comment:
Again, what Service types we care about? In all the Service type change case, we should always process the newPodSet?

Reply:
We essentially care about transitions from `non-nodeport -> nodeport` and vice-versa.
In cases of Service Type updates, I anticipate no change in the set of Pods (considering selectors to be identical at this moment), so we can just check the newPodSet for this, since old one must be the same.

Reply:
> In cases of Service Type updates, I anticipate no change in the set of Pods (considering selectors to be identical at this moment), so we can just check the newPodSet for this, since old one must be the same.

Can you clarify this? I don't think there is any guarantee that the Service Type update will be the only change in the Service spec. The selector may change as well. Even if the user first updated the Service Type, then updated the Selector separately, these 2 updates may be combined and the update handler may be called a single time.

Reply:
that is correct, the selector may change as well in the same update, in which case it should come in the following conditional, where it compares the selectors in the spec.